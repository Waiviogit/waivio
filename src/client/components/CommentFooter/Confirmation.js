import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import './Confirmation.less';

const Confirmation = ({ onConfirm, onCancel, isCashout, type }) => (
  <div className="Confirmation">
    {!isCashout && (
      <a role="presentation" onClick={onConfirm}>
        <i className="iconfont icon-success" />
        <FormattedMessage id={type} defaultMessage={type} />
      </a>
    )}
    <a role="presentation" onClick={onCancel}>
      <i className="iconfont icon-delete" />
      <FormattedMessage id="cancel" defaultMessage="Cancel" />
    </a>
  </div>
);

Confirmation.propTypes = {
  onConfirm: PropTypes.func,
  onCancel: PropTypes.func,
  isCashout: PropTypes.bool,
  type: PropTypes.string,
};

Confirmation.defaultProps = {
  onConfirm: () => {},
  onCancel: () => {},
  isCashout: false,
  type: 'Confirm',
};

export default Confirmation;
