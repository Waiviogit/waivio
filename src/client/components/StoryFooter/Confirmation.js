import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import './Confirmation.less';

const Confirmation = ({ onConfirm, onCancel, isCashout }) => (
  <div className="Confirmation">
    {!isCashout && (
      <a role="presentation" onClick={onConfirm}>
        <i className="iconfont icon-success" />
        <FormattedMessage id="confirm" defaultMessage="Confirm" />
      </a>
    )}
    <a role="presentation" onClick={onCancel}>
      <i className="iconfont icon-delete" />
      <FormattedMessage id="cancel" defaultMessage="Cancel" />
    </a>
  </div>
);

Confirmation.propTypes = {
  onConfirm: PropTypes.func,
  onCancel: PropTypes.func,
  isCashout: PropTypes.bool,
};

Confirmation.defaultProps = {
  onConfirm: () => {},
  onCancel: () => {},
  isCashout: false,
};

export default Confirmation;
