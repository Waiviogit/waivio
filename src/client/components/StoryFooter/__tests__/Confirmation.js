import React from 'react';
import { act } from 'react-dom/test-utils';

import { shallow, mount } from 'enzyme';
import Confirmation from '../Confirmation';

describe('<Confirmation />', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      onConfirm: () => {},
      onCancel: () => {},
      isCashout: false,
    };

    wrapper = mount(<Confirmation {...props} />);
    act(() => {
      wrapper.update();
    });
  });

  it('Should include Confirmation in className in tag div ', () => {
    expect(wrapper.find('div').props().className).toBe('Confirmation');
  });

  it('should render without exploding when post isn`t cashout', () => {
    wrapper = shallow(<Confirmation {...props} />);
    act(() => {
      wrapper.update();
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('should render without exploding when post is cashout', () => {
    wrapper = shallow(<Confirmation {...props} isCashout />);
    act(() => {
      wrapper.update();
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('Should not to be undefined component', () => {
    expect(wrapper).not.toBeUndefined();
  });
});
