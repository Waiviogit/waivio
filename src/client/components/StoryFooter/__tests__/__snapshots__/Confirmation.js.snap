// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Confirmation /> should render without exploding when post is cashout 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Confirmation
    isCashout={true}
    onCancel={[Function]}
    onConfirm={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        <a
          onClick={[Function]}
          role="presentation"
        >
          <i
            className="iconfont icon-delete"
          />
          <FormattedMessage
            defaultMessage="Cancel"
            id="cancel"
            values={Object {}}
          />
        </a>,
      ],
      "className": "Confirmation",
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <i
              className="iconfont icon-delete"
            />,
            <FormattedMessage
              defaultMessage="Cancel"
              id="cancel"
              values={Object {}}
            />,
          ],
          "onClick": [Function],
          "role": "presentation",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "iconfont icon-delete",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "defaultMessage": "Cancel",
              "id": "cancel",
              "values": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "a",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          <a
            onClick={[Function]}
            role="presentation"
          >
            <i
              className="iconfont icon-delete"
            />
            <FormattedMessage
              defaultMessage="Cancel"
              id="cancel"
              values={Object {}}
            />
          </a>,
        ],
        "className": "Confirmation",
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i
                className="iconfont icon-delete"
              />,
              <FormattedMessage
                defaultMessage="Cancel"
                id="cancel"
                values={Object {}}
              />,
            ],
            "onClick": [Function],
            "role": "presentation",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "iconfont icon-delete",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "defaultMessage": "Cancel",
                "id": "cancel",
                "values": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "a",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`<Confirmation /> should render without exploding when post isn\`t cashout 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Confirmation
    isCashout={false}
    onCancel={[Function]}
    onConfirm={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <a
          onClick={[Function]}
          role="presentation"
        >
          <i
            className="iconfont icon-success"
          />
          <FormattedMessage
            defaultMessage="Confirm"
            id="confirm"
            values={Object {}}
          />
        </a>,
        <a
          onClick={[Function]}
          role="presentation"
        >
          <i
            className="iconfont icon-delete"
          />
          <FormattedMessage
            defaultMessage="Cancel"
            id="cancel"
            values={Object {}}
          />
        </a>,
      ],
      "className": "Confirmation",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <i
              className="iconfont icon-success"
            />,
            <FormattedMessage
              defaultMessage="Confirm"
              id="confirm"
              values={Object {}}
            />,
          ],
          "onClick": [Function],
          "role": "presentation",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "iconfont icon-success",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "defaultMessage": "Confirm",
              "id": "confirm",
              "values": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "a",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <i
              className="iconfont icon-delete"
            />,
            <FormattedMessage
              defaultMessage="Cancel"
              id="cancel"
              values={Object {}}
            />,
          ],
          "onClick": [Function],
          "role": "presentation",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "iconfont icon-delete",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "defaultMessage": "Cancel",
              "id": "cancel",
              "values": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "a",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <a
            onClick={[Function]}
            role="presentation"
          >
            <i
              className="iconfont icon-success"
            />
            <FormattedMessage
              defaultMessage="Confirm"
              id="confirm"
              values={Object {}}
            />
          </a>,
          <a
            onClick={[Function]}
            role="presentation"
          >
            <i
              className="iconfont icon-delete"
            />
            <FormattedMessage
              defaultMessage="Cancel"
              id="cancel"
              values={Object {}}
            />
          </a>,
        ],
        "className": "Confirmation",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i
                className="iconfont icon-success"
              />,
              <FormattedMessage
                defaultMessage="Confirm"
                id="confirm"
                values={Object {}}
              />,
            ],
            "onClick": [Function],
            "role": "presentation",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "iconfont icon-success",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "defaultMessage": "Confirm",
                "id": "confirm",
                "values": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i
                className="iconfont icon-delete"
              />,
              <FormattedMessage
                defaultMessage="Cancel"
                id="cancel"
                values={Object {}}
              />,
            ],
            "onClick": [Function],
            "role": "presentation",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "iconfont icon-delete",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "defaultMessage": "Cancel",
                "id": "cancel",
                "values": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "a",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
